### PR TITLE
feat ShoppingCart model

### PR DESCRIPTION
**Pull request by Yam0r**

### CODE REVIEW

File: src/main/java/mate/academy/springintro/dto/shoppingcart/AddToCartRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+@Getter
+@Setter
+public class AddToCartRequestDto {
+    @NotNull
+    private Long bookId;
+
+    @Min(1)
+    private int quantity;
+}
+
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/springintro/dto/shoppingcart/AddToCartRequestDto.java]:
```suggestion
```

File: src/main/java/mate/academy/springintro/dto/shoppingcart/CartItemResponseDto.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+@Getter
+@Setter
+public class CartItemResponseDto {
+    private Long id;
+    private Long bookId;
+    private String bookTitle;
+    private int quantity;
+}
+
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/springintro/dto/shoppingcart/CartItemResponseDto.java]:
```suggestion
```
Leave only one empty line at the end of the file

File: src/main/java/mate/academy/springintro/dto/shoppingcart/ShoppingCartResponseDto.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+@Getter
+@Setter
+public class ShoppingCartResponseDto {
+    private Long id;
+    private Long userId;
+    private List<CartItemResponseDto> cartItems;
+
+    public ShoppingCartResponseDto(ShoppingCart shoppingCart) {
+    }
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/springintro/dto/shoppingcart/ShoppingCartResponseDto.java]: Redundant  constructor

File: src/main/java/mate/academy/springintro/model/CartItem.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+@Getter
+@Setter
+@Table(name = "cart_items")
+@RequiredArgsConstructor
+public class CartItem {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/springintro/model/CartItem.java]: use fetch type LAZY

Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/springintro/model/CartItem.java]:
```suggestion
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
```

File: src/main/java/mate/academy/springintro/model/CartItem.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    @JoinColumn(name = "shopping_cart_id", nullable = false)
+    private ShoppingCart shoppingCart;
+
+    @ManyToOne
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/springintro/model/CartItem.java]: the same

File: src/main/java/mate/academy/springintro/model/CartItem.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    @JoinColumn(name = "book_id", nullable = false)
+    private Book book;
+
+    @Column(nullable = false)
+    private int quantity;
+
+    public CartItem(ShoppingCart shoppingCart, Book book, int quantity) {
+    }
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/springintro/model/CartItem.java]: redundant constructor

File: src/main/java/mate/academy/springintro/model/CartItem.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+import jakarta.persistence.Table;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+
+@Entity
+@Getter
+@Setter
+@Table(name = "cart_items")
+@RequiredArgsConstructor
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/springintro/model/CartItem.java]: why do you need this annotation?

File: src/main/java/mate/academy/springintro/security/CustomUserDetails.java [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import mate.academy.springintro.model.User;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+@Getter
+@RequiredArgsConstructor
+public class CustomUserDetails implements UserDetails {
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/springintro/security/CustomUserDetails.java]: Why do you need it? Let’s  implement it in the User entity

File: src/main/java/mate/academy/springintro/security/CustomUserDetails.java [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    @Override
+    public boolean isCredentialsNonExpired() {
+        return true;
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return true;
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/springintro/security/CustomUserDetails.java]:
```suggestion
        return !isDeleted;
```

File: src/main/java/mate/academy/springintro/service/ShoppingCartService.java [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+    ShoppingCartResponseDto getCartForUser(Long userId);
+
+    ShoppingCartResponseDto addToCart(Long userId, AddToCartRequestDto requestDto);
+
+    ShoppingCartResponseDto updateCartItem(Long userId, Long cartItemId,
+                                           UpdateCartItemRequestDto requestDto);
+
+    void removeCartItem(Long userId, Long cartItemId);
+}
+
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/springintro/service/ShoppingCartService.java]: the same (fix it everywhere)

File: src/main/java/mate/academy/springintro/service/impl/ShoppingCartServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+import mate.academy.springintro.repository.CartItemRepository;
+import mate.academy.springintro.repository.ShoppingCartRepository;
+import mate.academy.springintro.repository.UserRepository;
+import mate.academy.springintro.service.ShoppingCartService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/springintro/service/impl/ShoppingCartServiceImpl.java]:
```suggestion
```

File: src/main/java/mate/academy/springintro/model/ShoppingCart.java [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+@Setter
+@SQLDelete(sql = "UPDATE shopping_carts SET is_deleted = true WHERE id = ?")
+@Where(clause = "is_deleted = FALSE")
+@Table(name = "shopping_carts")
+public class ShoppingCart {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @OneToOne
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/springintro/model/ShoppingCart.java:29]: you can use MapsId

File: src/main/java/mate/academy/springintro/service/impl/ShoppingCartServiceImpl.java [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+    private final UserRepository userRepository;
+    private final CartItemMapper cartItemMapper;
+
+    @Override
+    public ShoppingCartResponseDto getCartForUser(Long userId) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(userId)
+                .orElseThrow(() -> new RuntimeException("Shopping Cart not found for user "
+                        + userId));
+
+        return new ShoppingCartResponseDto(shoppingCart);
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/springintro/service/impl/ShoppingCartServiceImpl.java]: What are you return? Let’s  use a mapper

File: src/main/java/mate/academy/springintro/service/impl/ShoppingCartServiceImpl.java [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+    @Override
+    public ShoppingCartResponseDto addToCart(Long userId, AddToCartRequestDto requestDto) {
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new RuntimeException("User not found with id " + userId));
+
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(userId)
+                .orElseGet(() -> createShoppingCartForUser(user));
+
+        Book book = new Book(); // Assume you have a way to get the book by ID
+        book.setId(requestDto.getBookId());
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/springintro/service/impl/ShoppingCartServiceImpl.java]: get the book by ID from DB

File: src/main/java/mate/academy/springintro/service/impl/ShoppingCartServiceImpl.java [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+                                                  Long cartItemId,
+                                                  UpdateCartItemRequestDto requestDto) {
+        CartItem cartItem = cartItemRepository.findById(cartItemId)
+                .orElseThrow(() -> new RuntimeException("CartItem not found with id "
+                        + cartItemId));
+
+        cartItem.setQuantity(requestDto.getQuantity());
+        cartItemRepository.save(cartItem);
+
+        return new ShoppingCartResponseDto(cartItem.getShoppingCart());
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/springintro/service/impl/ShoppingCartServiceImpl.java]: the same, use mapper

File: src/main/java/mate/academy/springintro/service/impl/ShoppingCartServiceImpl.java [hunk 16]
```diff
@@ -0,0 +1,86 @@
...
+    @Override
+    public void removeCartItem(Long userId, Long cartItemId) {
+        CartItem cartItem = cartItemRepository.findById(cartItemId)
+                .orElseThrow(() -> new RuntimeException("CartItem not found with id "
+                        + cartItemId));
+
+        cartItemRepository.delete(cartItem);
+    }
+
+    private ShoppingCart createShoppingCartForUser(User user) {
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/springintro/service/impl/ShoppingCartServiceImpl.java]: you should call this logic in the registration method

File: src/main/resources/db/changelog/changes/create-cart-items-table.yaml [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+            baseColumnNames: shopping_cart_id
+            referencedTableName: shopping_carts
+            referencedColumnNames: id
+            constraintName: fk_cart_items_shopping_cart
+        - addForeignKeyConstraint:
+            baseTableName: cart_items
+            baseColumnNames: book_id
+            referencedTableName: books
+            referencedColumnNames: id
+            constraintName: fk_cart_items_book
```
Comment (liliia-ponomarenko) [inline src/main/resources/db/changelog/changes/create-cart-items-table.yaml]: add one empty line (check other files)

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 18]
```diff
@@ -13,4 +13,7 @@ databaseChangeLog:
       file: db/changelog/changes/create-users-roles-table.yaml
   - include:
       file: db/changelog/changes/insert-category.yaml
-
+#  - include:
+#      file: db/changelog/changes/create-shopping-carts-table.yaml
+#  - include:
+#      file: db/changelog/changes/create-cart-items-table.yaml
```
Comment (liliia-ponomarenko) [inline src/main/resources/db/changelog/db.changelog-master.yaml]: uncomment new changesets

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s  improve your solution;)

File: src/main/java/mate/academy/springintro/mapper/ShoppingCartMapper.java [hunk 19]
```diff
@@ -0,0 +1,20 @@
...
+@Mapper(config = MapperConfig.class)
+public interface ShoppingCartMapper {
+    @Mapping(target = "userId", source = "user.id")
+    @Mapping(target = "cartItems", source = "cartItems")
+    ShoppingCartResponseDto toShoppingCartResponseDto(ShoppingCart shoppingCart);
+
+    @Mapping(target = "bookId", source = "book.id")
+    CartItemResponseDto toCartItemResponseDto(CartItem cartItem);
+}
+
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/springintro/mapper/ShoppingCartMapper.java]:
```suggestion
```

File: src/main/java/mate/academy/springintro/model/CartItem.java [hunk 20]
```diff
@@ -0,0 +1,43 @@
...
+    @Column(nullable = false)
+    private int quantity;
+
+    public CartItem(ShoppingCart shoppingCart, Book book, int quantity) {
+        this.shoppingCart = shoppingCart;
+        this.book = book;
+        this.quantity = quantity;
+    }
+}
+
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/springintro/model/CartItem.java]:
```suggestion
```
Only one empty line at the end of the file

File: src/main/java/mate/academy/springintro/service/impl/UserServiceImpl.java [hunk 21]
```diff
@@ -40,11 +54,90 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto)
...
     }
+
+    @Override
+    public ShoppingCartResponseDto getCartForUser(Long userId) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(userId)
+                .orElseThrow(() -> new RuntimeException("Shopping Cart not found for user "
+                        + userId));
+        return toShoppingCartDto(shoppingCart);
+    }
+
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/springintro/service/impl/UserServiceImpl.java]: Why do you move ShoppingCart logic to UserService? Let’s  move it to the separate ShoppingCartService class

File: src/main/java/mate/academy/springintro/service/impl/UserServiceImpl.java [hunk 22]
```diff
@@ -40,11 +54,90 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto)
...
+        ShoppingCartResponseDto dto = new ShoppingCartResponseDto();
+        dto.setId(shoppingCart.getId());
+        dto.setUserId(shoppingCart.getUser().getId());
+        dto.setCartItems(
+                shoppingCart.getCartItems().stream()
+                        .map(this::toCartItemDto)
+                        .toList()
+        );
+        return dto;
+    }
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/springintro/service/impl/UserServiceImpl.java]: use mapper

File: src/main/java/mate/academy/springintro/service/impl/UserServiceImpl.java [hunk 23]
```diff
@@ -40,11 +54,90 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto)
...
+
+    private CartItemResponseDto toCartItemDto(CartItem cartItem) {
+        CartItemResponseDto dto = new CartItemResponseDto();
+        dto.setId(cartItem.getId());
+        dto.setBookId(cartItem.getBook().getId());
+        dto.setQuantity(cartItem.getQuantity());
+        return dto;
+    }
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/springintro/service/impl/UserServiceImpl.java]: the same

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 24]
```diff
@@ -13,4 +13,3 @@ databaseChangeLog:
       file: db/changelog/changes/create-users-roles-table.yaml
   - include:
       file: db/changelog/changes/insert-category.yaml
-
```
Comment (liliia-ponomarenko) [inline src/main/resources/db/changelog/db.changelog-master.yaml:16]: include new changesets to  changelog-master.yaml

File: src/main/java/mate/academy/springintro/model/ShoppingCart.java (hunk 12 above)
Comment (liliia-ponomarenko) [reply src/main/java/mate/academy/springintro/model/ShoppingCart.java:29]: not fixed

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s  fix your solution ;)

File: src/main/java/mate/academy/springintro/controller/ShoppingCartController.java [hunk 25]
```diff
@@ -0,0 +1,68 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/cart")
+@RequiredArgsConstructor
+@Tag(name = "Shopping Cart Management")
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+
+    @GetMapping
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/controller/ShoppingCartController.java:38]: add swagger annotations where are needed

File: src/main/java/mate/academy/springintro/dto/shoppingcart/AddToCartRequestDto.java [hunk 26]
```diff
@@ -0,0 +1,16 @@
...
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+public class AddToCartRequestDto {
+    @NotNull
+    private Long bookId;
+
+    @Min(1)
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/dto/shoppingcart/AddToCartRequestDto.java]:
```suggestion
    @NotNull
    @Positive
    private Long bookId;

    @Positive
```

File: src/main/java/mate/academy/springintro/dto/shoppingcart/UpdateCartItemRequestDto.java [hunk 27]
```diff
@@ -0,0 +1,12 @@
+package mate.academy.springintro.dto.shoppingcart;
+
+import javax.validation.constraints.Min;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+public class UpdateCartItemRequestDto {
+    @Min(1)
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/dto/shoppingcart/UpdateCartItemRequestDto.java]:
```suggestion
    @Positive
```

File: src/main/java/mate/academy/springintro/model/User.java [hunk 28]
```diff
@@ -58,7 +60,9 @@ public class User implements UserDetails {
 
     @Override
     public Collection<? extends GrantedAuthority> getAuthorities() {
-        return roles;
+        return roles.stream()
+                .map(role -> new SimpleGrantedAuthority(role.getRole().name()))
+                .collect(Collectors.toList());
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/model/User.java]:
```suggestion
        return roles;
```

File: src/main/java/mate/academy/springintro/service/impl/ShoppingCartServiceImpl.java [hunk 29]
```diff
@@ -0,0 +1,82 @@
...
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final BookRepository bookRepository;
+    private final UserRepository userRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+
+    @Override
+    public ShoppingCartResponseDto getCartForUser(Long userId) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(userId)
+                .orElseThrow(() -> new RuntimeException("Shopping Cart not found for user "
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/service/impl/ShoppingCartServiceImpl.java]:
```suggestion
                .orElseThrow(() -> new EntityNotFoundException("Shopping Cart not found for user "
```

File: src/main/java/mate/academy/springintro/service/impl/ShoppingCartServiceImpl.java [hunk 30]
```diff
@@ -0,0 +1,82 @@
...
+                        + userId));
+        return shoppingCartMapper.toShoppingCartResponseDto(shoppingCart);
+    }
+
+    @Override
+    @Transactional
+    public ShoppingCartResponseDto addToCart(Long userId, AddToCartRequestDto requestDto) {
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new RuntimeException("User not found with id " + userId));
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/service/impl/ShoppingCartServiceImpl.java]: check other places

```suggestion
                .orElseThrow(() -> new EntityNotFoundException("User not found with id " + userId));
```

File: src/main/java/mate/academy/springintro/service/impl/ShoppingCartServiceImpl.java [hunk 31]
```diff
@@ -0,0 +1,82 @@
...
+    @Override
+    @Transactional
+    public void removeCartItem(Long userId, Long cartItemId) {
+        CartItem cartItem = cartItemRepository.findById(cartItemId)
+                .orElseThrow(() -> new RuntimeException("CartItem not found with id "
+                        + cartItemId));
+        cartItemRepository.delete(cartItem);
+    }
+
+    private ShoppingCart createShoppingCartForUser(User user) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/service/impl/ShoppingCartServiceImpl.java]:
```suggestion
    @Override
    public ShoppingCart createShoppingCartForUser(User user) {
```

File: src/main/java/mate/academy/springintro/service/impl/UserServiceImpl.java [hunk 32]
```diff
@@ -34,17 +36,18 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto)
...
-                .orElseThrow(() -> new EntityNotFoundException(String.format(NOT_FOUND_ROLE
-                        + roleName)));
-
+                .orElseThrow(() -> new EntityNotFoundException(String.format(NOT_FOUND_ROLE,
+                        roleName)));
         user.setRoles(Set.of(userRole));
         userRepository.save(user);
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCartRepository.save(shoppingCart);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/service/impl/UserServiceImpl.java:49]:
```suggestion
        shoppingCartService.createShoppingCartForUser(user);
```

File: src/main/java/mate/academy/springintro/service/ShoppingCartService.java [hunk 33]
```diff
@@ -0,0 +1,16 @@
...
+public interface ShoppingCartService {
+    ShoppingCartResponseDto getCartForUser(Long userId);
+
+    ShoppingCartResponseDto addToCart(Long userId, AddToCartRequestDto requestDto);
+
+    ShoppingCartResponseDto updateCartItem(Long userId, Long cartItemId,
+                                           UpdateCartItemRequestDto requestDto);
+
+    void removeCartItem(Long userId, Long cartItemId);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/service/ShoppingCartService.java:17]:
```suggestion
    void removeCartItem(Long userId, Long cartItemId);
    
    ShoppingCart createShoppingCartForUser(User user);
```

File: src/main/resources/db/changelog/changes/create-shopping-carts-table.yaml [hunk 34]
```diff
@@ -0,0 +1,27 @@
+databaseChangeLog:
+  - changeSet:
+      id: create-shopping-carts
+      author: mateacademy
+      changes:
+        - createTable:
+            tableName: shopping_carts
+            columns:
+              - column:
+                  name: id
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/create-shopping-carts-table.yaml:10]: remove, you are using `@MapsId` in entity

File: src/main/resources/db/changelog/changes/create-shopping-carts-table.yaml [hunk 35]
```diff
@@ -0,0 +1,27 @@
...
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: user_id
+                  type: bigint
+                  constraints:
+                    nullable: false
+                    unique: true
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/create-shopping-carts-table.yaml:21]:
```suggestion
                  constraints:
                    primaryKey: true
                    nullable: false
```

File: src/main/java/mate/academy/springintro/model/ShoppingCart.java [hunk 36]
```diff
@@ -0,0 +1,42 @@
...
+
+@Entity
+@Getter
+@Setter
+@SQLDelete(sql = "UPDATE shopping_carts SET is_deleted = true WHERE id = ?")
+@Where(clause = "is_deleted = FALSE")
+@Table(name = "shopping_carts")
+public class ShoppingCart {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/model/ShoppingCart.java]:
```suggestion
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/mate/academy/springintro/service/impl/UserServiceImpl.java (hunk 32 above)
Comment (liliia-ponomarenko) [reply src/main/java/mate/academy/springintro/service/impl/UserServiceImpl.java:49]: not fixed

File: src/main/java/mate/academy/springintro/service/impl/UserServiceImpl.java [hunk 37]
```diff
@@ -34,17 +36,18 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto)
...
         return userMapper.toUserResponse(user);
     }
 }
+
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/springintro/service/impl/UserServiceImpl.java:53]:
```suggestion
```
Leave only one empty line

File: src/main/resources/db/changelog/changes/create-shopping-carts-table.yaml [hunk 38]
```diff
@@ -0,0 +1,33 @@
...
+      id: create-shopping-carts
+      author: mateacademy
+      changes:
+        - createTable:
+            tableName: shopping_carts
+            columns:
+              - column:
+                  name: id
+                  type: bigint
+                  autoIncrement: true
```
Comment (liliia-ponomarenko) [inline src/main/resources/db/changelog/changes/create-shopping-carts-table.yaml:12]: using MapsId you can have only one column `id` or `user_id`

**Review by liliia-ponomarenko — APPROVED** [review]
Good job! But fix my comments before merging ;)

**Pull request merged**